snapcraft/application.py

```diff
@@ -111,13 +113,23 @@ def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Pat
         if project_dir is None:
             project_dir = pathlib.Path.cwd()
 
-        try:
+        with suppress(FileNotFoundError):
             return super()._resolve_project_path(project_dir / "snap")
-        except FileNotFoundError:
-            try:
-                return super()._resolve_project_path(project_dir)
-            except FileNotFoundError:
-                return super()._resolve_project_path(project_dir / "build-aux" / "snap")
+
+        with suppress(FileNotFoundError):
+            return super()._resolve_project_path(project_dir)
+
+        with suppress(FileNotFoundError):
+            return super()._resolve_project_path(project_dir / "build-aux" / "snap")
+
+        raise ProjectFileMissingError(
```

Comment:
This is tricky because now there's a disagreement between craft-app and snapcraft; in craft-app, this method raises `FileNotFoundError`, and there's code in Snapcraft that [relies on that](https://github.com/canonical/snapcraft/blob/feature/craft-application/snapcraft/application.py#L80) (I'm surprised it didn't break any tests).

Reply:
That because `ProjectFileMissingError` also inherited `FileNotFoundError`, so it still got caught.

Reply:
I see, but then isn't the `ProjectFileMissingError` error that craft-app raises in get_project() "swallowing" this one?

Reply:
Oh, it did swallowed this one.

Reply:
What do you think about overriding get_project(), instead of this _resolve_project_path()? It could look like this:

```python
def get_project(self, ...):
    try:
        return super().get_project(...)
    except ProjectFileMissingError:
        raise ProjectFileMissingError(...) from None